tests/test_dates.py

```diff
@@ -107,100 +107,112 @@ def test_find_gaps_overlaps(input, flag, expected):
+        (({'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb'}, None),
          {'begin_date': datetime.date(2019, 1, 1),
           'end_date': datetime.date(2020, 2, 29),
           'date': None
           }
          ),
-        ({'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb', 'end_day': 10},
+        (({'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb', 'end_day': 10}, None),
          {'begin_date': datetime.date(2019, 1, 1),
           'end_date': datetime.date(2020, 2, 10),
           'date': None
           }
          ),
-        ({'begin_date': '2020-05-09', 'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb',
-          'end_day': 10},
+        (({'begin_date': '2020-05-09', 'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb',
+          'end_day': 10}, None),
          {'begin_date': datetime.date(2020, 5, 9),
           'end_date': datetime.date(2020, 2, 10),
           'date': None
           }
          ),
-        ({'end_date': '2020-5-20', 'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb',
-          'end_day': 10},
+        (({'end_date': '2020-5-20', 'begin_year': 2019, 'end_year': 2020, 'end_month': 'Feb',
+          'end_day': 10}, None),
          {'begin_date': datetime.date(2019, 1, 1),
           'end_date': datetime.date(2020, 5, 20),
           'date': None
           }
          ),
-        ({'date': '2020-5-20', 'begin_year': 2019, 'end_year': 2020,
+        (({'date': '2020-5-20', 'begin_year': 2019, 'end_year': 2020,
           'end_month': 'Feb',
-          'end_day': 10},
+          'end_day': 10}, None),
          {'begin_date': datetime.date(2019, 1, 1),
           'end_date': datetime.date(2020, 2, 10),
           'date': datetime.date(2020, 5, 20)
           }
          ),
-        ({'date': datetime.date(2020, 5, 20), 'begin_year': 2019, 'end_year': 2020,
+        (({'date': datetime.date(2020, 5, 20), 'begin_year': 2019, 'end_year': 2020,
           'end_month': 'Feb',
-          'end_day': 10},
+          'end_day': 10}, None),
          {'begin_date': datetime.date(2019, 1, 1),
           'end_date': datetime.date(2020, 2, 10),
           'date': datetime.date(2020, 5, 20)
           }
          ),
-        ({'date': datetime.date(2020, 5, 20), 'begin_date': datetime.date(2015, 6, 8),
-          'end_date': datetime.date(2025, 10, 4)},
+        (({'date': datetime.date(2020, 5, 20), 'begin_date': datetime.date(2015, 6, 8),
+          'end_date': datetime.date(2025, 10, 4)}, None),
          {'begin_date': datetime.date(2015, 6, 8),
           'end_date': datetime.date(2025, 10, 4),
           'date': datetime.date(2020, 5, 20)
           }
          ),
+        (({'submission_day': 10, 'submission_year': 2020,
+          'submission_month': 'Feb'}, "submission"),
+         {'begin_date': datetime.date(2020, 2, 10),
+          'end_date': datetime.date(2020, 2, 10),
+          'date': datetime.date(2020, 2, 10)
```

Comment:
I wonder if this should return ``'submission_date':`` in the dict?  It could be confusing for people otherwise.